### DISCUSSION TITLE
Get screenshot not work

### DISCUSSION THREAD

**Discussion by AndresSantiago15** [appium/appium#19119] [Q&A]
Hi, 

I use appium 1.21.0 with python and Android. When i take a picture like driver.get_screenshot_as_file('file.png') appium give me a corrupted png. Any one can help me?

**Comment by portyu9** [comment]
Hi @AndresSantiago15 

`driver.get_screenshot_as_file('file.png')` is a valid way to save an Appium screenshot, so a corrupted PNG by itself is not enough to identify the root cause.

I would first determine whether the corruption comes from Appium/UiAutomator2 or from saving the file.

Try getting the raw screenshot and decoding it yourself:

```python
import base64

data = driver.get_screenshot_as_base64()

with open("file.png", "wb") as f:
    f.write(base64.b64decode(data))
```

If that file is also corrupted, check whether Android itself can produce a valid screenshot:

```bash
adb exec-out screencap -p > adb-screenshot.png
```

That separates the problem into three cases:

* `adb-screenshot.png` works, but Appium's screenshot is corrupt: investigate the Appium/UiAutomator2 server and check the Appium server log around the screenshot request.
* Both screenshots are bad: the problem is likely at the Android/device level.
* `get_screenshot_as_base64()` produces a valid PNG but `get_screenshot_as_file()` does not: investigate the Python client/version or local file handling.

Also, Appium `1.21.0` is very old now. I would reproduce the problem on a current supported Appium + UiAutomator2 driver before investigating an Appium 1.x-specific screenshot bug.

The Android version/device model and the Appium server log for the screenshot request would be the most useful information if it still fails after upgrading.

Please mark this answer as accepted if it helped, thank you.
